src/Generators/AbstractTagGenerator.php

```diff
@@ -261,4 +274,44 @@ protected function pushMethodTag($methodName, $tagString)
             $this->tags['methods'][$tagString] = $this->pushTagWithExistingComment('method', $tagString);
         }
     }
+
+    /**
+     * Factory method for easy instantiation.
+     * @param array<string, class-string<Tag>|Factory> $additionalTags
+     * @return DocBlockFactoryInterface
+     */
+    protected function createShortNameFactory(array $additionalTags = []): DocBlockFactory
+    {
+        $fqsenResolver = new ShortNameResolver();
+        $tagFactory = new StandardTagFactory($fqsenResolver);
+        $descriptionFactory = new DescriptionFactory($tagFactory);
+        $typeResolver = new TypeResolver($fqsenResolver);
+
+        $phpstanTagFactory = new AbstractPHPStanFactory(
+            new ParamFactory($typeResolver, $descriptionFactory),
+            new VarFactory($typeResolver, $descriptionFactory),
+            new ReturnFactory($typeResolver, $descriptionFactory),
+            new PropertyFactory($typeResolver, $descriptionFactory),
+            new PropertyReadFactory($typeResolver, $descriptionFactory),
+            new PropertyWriteFactory($typeResolver, $descriptionFactory),
+            new MethodFactory($typeResolver, $descriptionFactory)
+        );
+
+        $tagFactory->addService($descriptionFactory);
+        $tagFactory->addService($typeResolver);
+        $tagFactory->registerTagHandler('param', $phpstanTagFactory);
+        $tagFactory->registerTagHandler('var', $phpstanTagFactory);
+        $tagFactory->registerTagHandler('return', $phpstanTagFactory);
+        $tagFactory->registerTagHandler('property', $phpstanTagFactory);
+        $tagFactory->registerTagHandler('property-read', $phpstanTagFactory);
+        $tagFactory->registerTagHandler('property-write', $phpstanTagFactory);
+        $tagFactory->registerTagHandler('method', $phpstanTagFactory);
+
+        $docBlockFactory = new DocBlockFactory($descriptionFactory, $tagFactory);
+        foreach ($additionalTags as $tagName => $tagHandler) {
+            $docBlockFactory->registerTagHandler($tagName, $tagHandler);
+        }
+
+        return $docBlockFactory;
+    }
```

Comment:
Looks good, because the tests aren't running at the moment, I'll go with accepting at face-value